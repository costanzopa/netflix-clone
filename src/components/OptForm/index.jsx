export { default } from './OptForm.component';
