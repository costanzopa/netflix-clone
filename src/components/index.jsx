export { default as Jumbotron } from './Jumbotron';
export { default as App } from './App';
