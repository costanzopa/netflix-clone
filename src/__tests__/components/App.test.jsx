import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { App } from '../../components';
import { FirebaseContext } from '../../context/firebase';
import { firebase } from '../../lib/firabase.prod';
import { useAuthListener } from '../../hooks';

jest.mock('../../hooks');

describe('<App />', () => {
  it('renders the <App /> Home Page', () => {
    const { container } = render(
      <FirebaseContext.Provider value={{ firebase }}>
        <App />
      </FirebaseContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <App /> Sign in Page', () => {
    const { container, getByText } = render(
      <FirebaseContext.Provider value={{ firebase }}>
        <App />
      </FirebaseContext.Provider>
    );

    fireEvent.click(getByText('Sign In'));
    expect(getByText('Loading...')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <App /> Sign Up Page', async () => {
    const { container, getByText } = render(
      <FirebaseContext.Provider value={{ firebase }}>
        <App />
      </FirebaseContext.Provider>
    );
    fireEvent.click(getByText('Sign up now.'));
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <App /> Browse Page', async () => {
    useAuthListener.mockReturnValue({});
    const { container } = render(
      <FirebaseContext.Provider value={{ firebase }}>
        <App />
      </FirebaseContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
