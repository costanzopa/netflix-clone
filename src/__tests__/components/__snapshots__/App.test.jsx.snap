// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the <App /> Browse Page 1`] = `
<div>
  Loading...
</div>
`;

exports[`<App /> renders the <App /> Home Page 1`] = `
<div>
  Loading...
</div>
`;

exports[`<App /> renders the <App /> Sign Up Page 1`] = `
<div>
  Loading...
</div>
`;

exports[`<App /> renders the <App /> Sign in Page 1`] = `
<div>
  Loading...
</div>
`;
